src/tasks/component-installers/devfile-workspace-operator-installer.ts

```diff
@@ -0,0 +1,336 @@
+            configMapYaml.data['devworkspace.routing.cluster_host_suffix'] = flags.domain
+          }
+
+          let webHooksValue = 'false'
+          let routingClass = 'basic'
+          const isOpenShift = await this.kubeHelper.isOpenShift()
+          if (isOpenShift) {
+            routingClass = 'openshift-oauth'
+            webHooksValue = 'true'
+          }
+          configMapYaml.data['controller.webhooks.enabled'] = webHooksValue
+          configMapYaml.data['devworkspace.default_routing_class'] = routingClass
+          configMapYaml.data['tls.insecure_skip_verify'] = 'true'
+
+          const configMap = await this.kubeHelper.getConfigMap(configMapName, this.getNamespace())
+          if (configMap) {
+            task.title = `${task.title}...Already Exists. Replacing`
+            await this.kubeHelper.replaceNamespacedConfigMap(configMapName, this.getNamespace(), configMapYaml)
+          } else {
+            await this.kubeHelper.createNamespacedConfigMap(this.getNamespace(), configMapYaml)
+          }
+          task.title = `${task.title}...done.`
+        }
+      },
+      {
+        title: 'Create dev workspace controller',
+        task: async (_ctx: any, task: any) => {
+          const exists = await this.kubeHelper.deploymentExist('devworkspace-controller', this.getNamespace())
+          if (exists) {
+            task.title = `${task.title}...It already exists.`
+            return
+          }
+          const isOpenShift = await this.kubeHelper.isOpenShift()
+          const yamls: any[] = []
+          if (isOpenShift) {
+            const yamlControllerFile = path.join(this.getTemplatePath(), 'os', 'controller.yaml')
+            const rawYaml = await fs.readFile(yamlControllerFile, 'utf-8')
+            yaml.safeLoadAll(rawYaml, yaml => {
+              yamls.push(yaml)
+            })
+          } else {
+            const yamlControllerFile = path.join(this.getTemplatePath(), 'k8s', 'controller.yaml')
+            const rawYaml = await fs.readFile(yamlControllerFile, 'utf-8')
+            yaml.safeLoadAll(rawYaml, yaml => {
+              yamls.push(yaml)
+            })
+          }
+          await Promise.all(yamls.map(async yaml => {
+            if (yaml.kind === 'Deployment') {
+              // customize devworkspace controller image:
+              yaml.spec.template.spec.containers[0].image = flags['dev-workspace-controller-image']
+              return this.kubeHelper.createDeploymentFrom(yaml, this.getNamespace())
+            }
+            if (yaml.kind === 'Service') {
+              return this.kubeHelper.createServiceFrom(yaml, this.getNamespace())
+            }
+          }))
+
+          task.title = `${task.title}...done.`
+        }
+      },
+    ]
+  }
+
+  /**
+   * Returns list of tasks which uninstall dev-workspace.
+   */
+  getUninstallTasks(): ReadonlyArray<Listr.ListrTask> {
+    return [
+      {
+        title: 'Delete all deployments for DevWorkspace controller',
```

Comment:
I'm not sure that user will be happy if we will remove all deployments and services from his namespace...

Reply:
But maybe for first iteration it's not very important. But it would be nice to have an issue for enchantments.

Reply:
It's a copy from what is done for che
https://github.com/che-incubator/chectl/blob/master/src/tasks/che.ts#L338-L349